test/testdrive/update.td

```diff
@@ -168,3 +168,37 @@ DELETE FROM c WHERE a < 3;
 
 > SELECT a FROM c;
 3
+
```

Comment:
I'd add a test to verify that lateral references to the main table from the USING clause are not allowed, like:

```
asenac=> delete from t1 using lateral (select t1.f1 from t2) t3 where t3.f1 = t1.f1;
ERROR:  invalid reference to FROM-clause entry for table "t1"
LINE 1: delete from t1 using lateral (select t1.f1 from t2) t3 where...
                                             ^
HINT:  There is an entry for table "t1", but it cannot be referenced from this part of the query.
asenac=> 
```